Woof/WoofTests/ViewModel/LoginViewModelTests/LoginViewModelSetUserRoleTests.swift

```diff
@@ -0,0 +1,39 @@
+import XCTest
+
+final class LoginViewModelSetUserRoleTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.testPreferencesStorage
+    private let testLoginViewModel = LoginViewModel()
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testMethodExistsInAPI() {
+        testLoginViewModel.set(.sitter)
+    }
+
+    func testSetsEveryPossibleRole() {
+        Role.allCases.forEach {
+            // Given When
+            testLoginViewModel.set($0)
+            
+            // Then
+            XCTAssertEqual($0, PreferencesHandler.getUserRole())
```

Comment:
nit:
```suggestion
            XCTAssertEqual(PreferencesHandler.getUserRole(), $0)
```

Reply:
Ok, thx.
In the future I will use this option because I refactored the code of this test.